src/display_controller.py

```
@@ -2507,6 +2499,160 @@ def _cleanup_expired_wifi_status(self):
+                self.available_modes.remove(mode)
+            self.plugin_modes.pop(mode, None)
+            self.mode_to_plugin_id.pop(mode, None)
+
+        # Unsubscribe the plugin's config-change callback. Pop only after the
+        # unsubscribe attempt so a failed unsubscribe doesn't lose our only
+        # reference to the callback before config_service has processed it.
+        callback = self._plugin_config_callbacks.get(plugin_id)
+        if callback is not None and hasattr(self, 'config_service'):
+            try:
+                self.config_service.unsubscribe(callback, plugin_id=plugin_id)
+            except Exception as e:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)